Assets/LeapMotionModules/Interaction/InteractionHand.cs

```diff
@@ -76,26 +86,73 @@ private struct HoverCheckResults {
     private HashSet<InteractionBehaviourBase> _hoveredLastFrame = new HashSet<InteractionBehaviourBase>();
 
     private HashSet<InteractionBehaviourBase> _hoverableCache = new HashSet<InteractionBehaviourBase>();
+    public InteractionBehaviourBase[] tempPerFingerHovered = new InteractionBehaviourBase[3];
+
     private HoverCheckResults CheckHoverForHand(Hand hand, HashSet<InteractionBehaviourBase> hoverCandidates) {
       _hoverableCache.Clear();
+      Array.Clear(tempPerFingerHovered, 0, tempPerFingerHovered.Length);
 
       HoverCheckResults results = new HoverCheckResults() {
         hovered = _hoverableCache,
+        perFingerHovered = tempPerFingerHovered,
         primaryHovered = null,
         primaryHoveredDistance = float.PositiveInfinity,
         checkedHand = hand
       };
 
-      foreach (var interactionObj in hoverCandidates) {
-        if (interactionObj.ignoreHover) continue;
-        CheckHoverForElement(hand, interactionObj, ref results);
+      //Loop through all the fingers (that we care about)
+      if (hand != null) {
+        float leastHandDistance = float.PositiveInfinity;
+        for (int i = 0; i < 3; i++) {
```

Comment:
I had to do some digging to figure out which of the fingers this referred to, it would be nice if you had a list of FingerType's as a constant somewhere that you could reference instead of using the finger indexes directly.